app/Services/EDocument/Imports/MindeeEDocument.php

```diff
@@ -0,0 +1,176 @@
+                $vendor->phone = $prediction->supplierPhoneNumber;
+                // $vendor->address1 = $address_1; // TODO: we only have the full address string from mindee returned
+                // $vendor->address2 = $address_2;
+                // $vendor->city = $city;
+                // $vendor->postal_code = $postcode;
+                $country = app('countries')->first(function ($c) use ($country) {
+                    return $c->iso_3166_2 == $country || $c->iso_3166_3 == $country;
+                });
+                if ($country)
+                    $vendor->country_id = $country->id;
+
+                $vendor->save();
+
+                if ($prediction->supplierEmail) {
+                    $vendor_contact = VendorContactFactory::create($user->company()->id, $user->id);
+                    $vendor_contact->vendor_id = $vendor->id;
+                    $vendor_contact->email = $prediction->supplierEmail;
+                    $vendor_contact->save();
+                }
+
+                $expense->vendor_id = $vendor->id;
+            }
+            $expense->transaction_reference = $documentno;
+        } else {
+            // The document exists as an expense
+            // Handle accordingly
+            nlog("Document already exists");
+            $expense->private_notes = $expense->private_notes . ctrans("texts.edocument_import_already_exists", ["date" => time()]);
+        }
+        $expense->save();
+        return $expense;
+    }
+
+    private function checkLimits()
+    {
+        $user = auth()->user();
+
+        Cache::add('mindeeTotalDailyRequests', 0, now()->endOfDay());
+        Cache::add('mindeeTotalMonthlyRequests', 0, now()->endOfMonth());
+        Cache::add('mindeeAccountDailyRequests' . $user->company->account->id, 0, now()->endOfDay());
+        Cache::add('mindeeAccountMonthlyRequests' . $user->company->account->id, 0, now()->endOfMonth());
+        if (config('services.mindee.daily_limit') != 0 && Cache::get('mindeeTotalDailyRequests') > config('services.mindee.daily_limit'))
+            throw new Exception('Mindee daily limit reached');
+        if (config('services.mindee.monthly_limit') != 0 && Cache::get('mindeeTotalMonthlyRequests') > config('services.mindee.monthly_limit'))
+            throw new Exception('Mindee monthly limit reached');
+        if (config('services.mindee.account_daily_limit') != 0 && Cache::get('mindeeAccountDailyRequests' . $user->company->account->id) > config('services.mindee.account_daily_limit'))
+            throw new Exception('Mindee daily limit reached for account: ' . $user->company->account->id);
+        if (config('services.mindee.account_monthly_limit') != 0 && Cache::get('mindeeAccountMonthlyRequests' . $user->company->account->id) > config('services.mindee.account_monthly_limit'))
+            throw new Exception('Mindee monthly limit reached for account: ' . $user->company->account->id);
```

Comment:
@turbo124 please review the limits section in detail, because this is important to reducs exesive use and costs!